crates/recursor/src/lib.rs

```diff
@@ -85,6 +90,77 @@ impl DnssecPolicy {
     }
 }
 
+/// An IPv4/IPv6 access control set.  This mainly hides the complexity of supporting v4 and v6
+/// addresses concurrently in a given set.  The AccessControlSet differs from a typical Access
+/// Control List in that there is no order.  The access semantics are:
+///
+/// | Present in allow list | Present in deny list |  Result  |
+/// |-----------------------|----------------------|----------|
+/// |                  true |                false |  allowed |
+/// |                  true |                 true |  allowed |
+/// |                 false |                false |  allowed |
+/// |                 false |                 true |   denied |
+#[derive(Clone)]
+pub(crate) struct AccessControlSet {
+    v4_allow: PrefixSet<Ipv4Net>,
+    v4_deny: PrefixSet<Ipv4Net>,
+    v6_allow: PrefixSet<Ipv6Net>,
+    v6_deny: PrefixSet<Ipv6Net>,
+}
+
+impl AccessControlSet {
+    pub(crate) fn new(name: &'static str, allow: &[IpNet], deny: &[IpNet]) -> Self {
```

Comment:
WDYT about using more of a builder-ish API with fns like `allow(&[IpNet])` and `deny(&[IpNet])`. Having two equivalent type params side-by side in `new()` makes it easy to mix them up and hard to tell which is which at the call-site.